### ISSUE TITLE
Quoting identifiers in with statement can fail

### ISSUE THREAD

**Issue by leocrawford** [neo4j/neo4j#4352]
The following query fails against neo 2.14 and 2.17 (and probably others) 

```
 MATCH (`a`)  WITH `a`  RETURN *
```

The exception contains  `Unknown identifier `a`.` Replacing it with any of the following works

```
MATCH (a)  WITH a  RETURN *
MATCH (`a`)  WITH `a` as a  RETURN *
MATCH (`a`)  WITH `a` as b  RETURN *
MATCH (`a`)  WITH `a` as `a`  RETURN *
```

**Comment by cleishm** [comment]
Hey @leocrawford,

Thanks for your report! I've confirmed that we've resolved this in Neo4j 2.2.0, so perhaps upgrade?

Cheers,
Chris

**Comment by leocrawford** [comment]
Thanks. Will do. 

Cheers
Leo
